### PR TITLE
feat(huginn_bigcommerce_product_agent.rb): disambiguate products by listing product types in name

### PR DESCRIPTION
**Pull request by John-Rabaey**
@mrogers-5s Would you please review?

### CODE REVIEW

**Comment by John-Rabaey** [comment]
https://trello.com/c/9GKlUygF/208-product-sync-fix-multiple-products-with-the-same-name

File: lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb [hunk 1]
```diff
@@ -143,6 +143,16 @@ def handle_variants(event)
                     product['name'] = "#{product['name']} (Digital)"
                 end
 
+                # Ignatius Press -- some products have the same name and must be disambiguated.
+                # ...by adding a list of the product types (hardback, paperback, etc.) to their names
+                if options['should_disambiguate'] == "true"
```
Comment (mrogers-5s) [inline lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb:148]: @John-Rabaey The logic here looks good, but I would do a couple things:

*  The option should be a true boolean value, and not a string of "true"
*  You should add validation for this option
   *  It should be _optional_ (evaluate to _false_ if not provided)
   *   If provided, it should be a boolean

(There's a boolean option example here: https://github.com/5-stones/huginn_record_link_agent/blob/master/lib/huginn_record_link_agent/record_link_agent.rb#L222)

Comment (John-Rabaey) [reply lib/huginn_bigcommerce_product_agent/bigcommerce_product_agent.rb]: Done! Thanks for the help!

**Comment by John-Rabaey** [comment]
@mrogers-5s How's this look?

**Comment by John-Rabaey** [comment]
@Jhalter5Stones Mike is out of office this week. Would you be willing to give this a quick look?

**Comment by Jhalter5Stones** [comment]
@John-Rabaey I think this looks good.

**Comment by John-Rabaey** [comment]
@Ken-Michalak Would you please review?

**Pull request merged**